ReverseTicTacToe/Logic/Game.cs

```diff
@@ -5,60 +5,308 @@
-            m_Player1Char = k_DefaultPlayer1Char;
-            m_Player2Char = k_DefaultPlayer2Char;
-            m_CurrentTurn = k_DefaultFirstTurn;
+            m_Player1Char = s_DefaultPlayer1Char;
+            m_Player2Char = s_DefaultPlayer2Char;
+            m_CurrentTurn = s_DefaultFirstTurn;
+            m_EmptySlotChar = s_DefaultEmptySlotChar;
             m_GameBoard = new char[m_BoardDimension, m_BoardDimension];
-            m_EmptySlotChar = k_DefaultEmptySlotChar;
+            ClearBoard();
+            m_Player1Score = 0;
+            m_Player2Score = 0;
+            m_GameStatus = eGameStatus.ClearBoard;
+            m_FilledSlotsCounter = 0;
 
-            for(int i = 0; i < m_BoardDimension; i++)
+        }
+
+        public int BoardSize
+        {
+            get
+            {
+                return m_BoardDimension;
+            }
+            set
             {
-                for(int j = 0; j < m_BoardDimension; j++)
+                if(m_GameStatus == eGameStatus.ClearBoard && IsInRange(value, s_I_GameBoardMinimumDimension,
+                       s_GameBoardMaximumDimension ))
+                {
+                    m_BoardDimension = value;
+                    m_GameBoard = new char[m_BoardDimension, m_BoardDimension];
+                    ClearBoard();
+                }
+
+            }
+        }
+
+        public static bool IsInRange(int i_Num, int i_Min, int i_Max)
+        {
+            return i_Num <= i_Max && i_Num >= i_Min;
+        }
+
+        public bool IsSlotIndexWithinBounds(int i_Row, int i_Col)
+        {
+            return (IsInRange(i_Row, 1, m_BoardDimension) && IsInRange(i_Col, 1, m_BoardDimension));
+        }
+
+
+
+        public bool AttemptMove(int i_Row, int i_Col, out eLastMoveStatus o_MoveStatus)
+        {
+            o_MoveStatus = eLastMoveStatus.Good;
+            char charToInsert = (m_CurrentTurn == eTurns.Player1) ? m_Player1Char : m_Player2Char;
+            const bool v_MoveMadeSuccessfully = true;
+            bool moveMade = !v_MoveMadeSuccessfully;
+
+            if(!IsSlotIndexWithinBounds(i_Row, i_Col))
+            {
+                o_MoveStatus = eLastMoveStatus.FailedOutOfBounds;
+            }
+            else if(IsGameOver())
+            {
+                o_MoveStatus = eLastMoveStatus.FailedGameOver;
+            }
+            else if(!IsSlotFree(i_Row, i_Col))
+            {
+                o_MoveStatus = eLastMoveStatus.FailedSlotTaken;
+            }
+            else
+            {
+                o_MoveStatus = eLastMoveStatus.Good;
+                makeMove(i_Row, i_Col);
+                moveMade = v_MoveMadeSuccessfully;
+            }
+
+            return moveMade;
+        }
+
+        //This functions assumes the move is legal and can be done.
+        private void makeMove(int i_Row, int i_Col)
+        {
+            m_GameBoard[i_Row - 1, i_Col - 1] = (m_CurrentTurn == eTurns.Player1) ? m_Player1Char : m_Player2Char;
```

Comment:
You got the 'charToInsert' in AttemptMove, maybe pass it to 'makeMove' as an argument to prevent code duplication

Reply:
Good point,  i think what you suggested might make the method "makeMove" a little bit less independent, and subject to mistakes. But i did fix that code duplication.